internal/rest/rest_operation.go

```diff
@@ -0,0 +1,248 @@
+	supportedModelTypes = map[string]modelSchema{
+		"VirtualService": virtualServiceModel,
+		"PoolGroup":      poolGroupModel,
+	}
+
+	// List of Objects for which we would handle error
+	supportedObjForError = map[string]struct{}{
+		"Pool":                 {},
+		"VsVip":                {},
+		"SSLKeyAndCertificate": {},
+	}
+)
+
+type poolGroupSchema struct {
+	Type   string
+	refMap map[string]func(*utils.RestOp, string) bool
+}
+
+func (v *poolGroupSchema) GetType() string {
+	return v.Type
+}
+func (v *poolGroupSchema) RemoveRef(refType string) func(*utils.RestOp, string) bool {
+	return v.refMap[refType]
+}
+
+func initPGModel() *poolGroupSchema {
+	pg := poolGroupSchema{}
+	pg.Type = "PoolGroup"
+	pg.refMap = map[string]func(*utils.RestOp, string) bool{
+		"Pool": pg.removePoolRef,
+	}
+	return &pg
+}
+
+func (v *poolGroupSchema) removePoolRef(op *utils.RestOp, objRef string) bool {
+	pg, ok := op.Obj.(*avimodels.PoolGroup)
+	if !ok {
+		utils.AviLog.Infof("Failed to remove Pool ref, object is not of type PoolGroup")
+		return false
+	}
+
+	for i := range pg.Members {
+		if strings.EqualFold(*pg.Members[i].PoolRef, objRef) {
+			pg.Members = append(pg.Members[:i], pg.Members[i+1:]...)
+			utils.AviLog.Infof("Successfully removed pool ref %s from PoolGroup: %s", objRef, *pg.Name)
+			break
+		}
+	}
+	op.Obj = pg
+	return true
+}
+
+type virtualserviceSchema struct {
+	Type   string
+	refMap map[string]func(*utils.RestOp, string) bool
+}
+
+func (v *virtualserviceSchema) GetType() string {
+	return v.Type
+}
+
+func initVSModel() *virtualserviceSchema {
+	vs := virtualserviceSchema{}
+	vs.Type = "VirtualService"
+	vs.refMap = map[string]func(*utils.RestOp, string) bool{
+		"SSLKeyAndCertificate": vs.removeCertRef,
+		"Pool":                 vs.removePoolRef,
+		"VsVip":                vs.removeVsVipRef,
+	}
+	return &vs
+}
+
+func (v *virtualserviceSchema) RemoveRef(refType string) func(*utils.RestOp, string) bool {
+	return v.refMap[refType]
+}
+
+func (v *virtualserviceSchema) removeCertRef(op *utils.RestOp, objRef string) bool {
+	vs, ok := op.Obj.(*avimodels.VirtualService)
+	if !ok {
+		utils.AviLog.Infof("Failed to remove SSL Cert ref, object is not of type Virtualservice")
+		return false
+	}
+	for i, v := range vs.SslKeyAndCertificateRefs {
+		if strings.EqualFold(v, objRef) {
+			vs.SslKeyAndCertificateRefs = append(vs.SslKeyAndCertificateRefs[:i], vs.SslKeyAndCertificateRefs[i+1:]...)
+			utils.AviLog.Infof("Successfully removed SSl Cert ref %s from VS: %s", objRef, *vs.Name)
+		}
+	}
+	op.Obj = vs
+	return true
+}
+
+func (v *virtualserviceSchema) removePoolRef(op *utils.RestOp, objRef string) bool {
+	vs, ok := op.Obj.(*avimodels.VirtualService)
+	if !ok {
+		utils.AviLog.Infof("Failed to remove Pool ref, object is not of type Virtualservice")
+		return false
+	}
+	if strings.EqualFold(*vs.PoolRef, objRef) {
+		vs.PoolRef = nil
+	}
+	for i := range vs.ServicePoolSelect {
```

Comment:
I don't think we should have ServicePoolSelect anymore. It should be l4policyset.

Reply:
Missed it, thanks. I have added one schema for L4PolicySet